mathesar_ui/src/stores/table-data/selection.ts

```diff
@@ -79,6 +87,30 @@ export class Selection {
     document.addEventListener('mouseup', () => {
       this.onEndSelection();
     });
+
+    // Keep active cell and selected cell in sync
+    this.display.activeCell.subscribe((activeCell) => {
```

Comment:
The `subscribe` function returns an `Unsubscriber` function, and to avoid memory leaks it's best-practice to find an appropriate place to call that Unsubscriber function. [Here](https://github.com/centerofci/mathesar/blob/e0ba80ba5abc495698f6f808dbbdf167784551a2/mathesar_ui/src/stores/table-data/records.ts#L307) is an example of how we store that Unsubscriber function and then [call it](https://github.com/centerofci/mathesar/blob/e0ba80ba5abc495698f6f808dbbdf167784551a2/mathesar_ui/src/stores/table-data/records.ts#L653) from a destroy method that [gets called](https://github.com/centerofci/mathesar/blob/e0ba80ba5abc495698f6f808dbbdf167784551a2/mathesar_ui/src/stores/table-data/tabularData.ts#L119) from `TabularData.destroy`. Can you implement the same pattern here? You'll need a new property on `Selection` to store the Unsubscriber function. Then you'll need a `destroy` method. Then you'll need to call `this.selection.destroy()` from `TabularData.destroy`

Reply:
@seancolsen makes sense. Done.